### PR TITLE
Transaction Header: Created a new Component Transaction Header.

### PR DESCRIPTION
**Pull request by NileshSingh9245**
### Release

1.0.0
https://blackminystudio.atlassian.net/browse/TPA-59

### Description

- Created a component Transaction Header

### Screenshots

Screenshot 2024-12-15 195409
---------------------------------------------------------------------------------------------------------------------------------------
Screenshot 2024-12-15 195430

### Tips

Use this code in main.dart for testing purpose: 
```Dart
class HomeScreen extends StatelessWidget {
  @override
  Widget build(BuildContext context) {
    final theme = Theme.of(context);
    return Scaffold(
      backgroundColor: theme.colors.light,
      body: Center(
        child: TransactionHeader(
          value: '3',
          onSeeAllPressed: () {
            print('See All');
          },
        ),
      ),
    );
  }
}
```

---

### IMPORTANT:

- (REQUIRED) Coverage Screenshots for the tests written
- (REQUIRED) UI Screenshots for the UI implementation written

### Author preparation checklist

- [x] Current branch is rebased with latest 'Staging' branch
- [x] PR is targeting only one issue
- [x] Commit messages are descriptive, consistent, and follow the project's guidelines.
- [ ] Branch name matches with at least `<jira link reference>-<short change description>`
- [x] Primary reviewer is set as assignee to the PR
- [x] No unnecessary changes on `pubspec.lock`
- [x] All added TODOs have associated Github issue references
- [x] \*.iml and, pubsepc_overrides.yaml are not checkedin
- [x] All hard coded strings are present in constants file

### Primary reviewer checklist

- [x] All items in the author checklist is checked or not applicable
- [x] All files including .lock and assets are reviewed
- [x] Reasonable UI flows/cases related to the change are manually tested
- [x] No commented-out code or unnecessary debug statements.
- [x] No spelling or grammatical errors in code comments or documentation.
- [x] No sensitive data (e.g., API keys, credentials) is hardcoded or exposed.
- [x] The branch is up-to-date with the target branch (e.g., staging).
- [ ] The code adheres to established coding standards.

### Pre-merge checklist

- [ ] Two approvals received. (Changes affecting automatic PR verification, like new linter rules, formatting rules, Github workflow configurations)

---

### CODE REVIEW

File: lib/widgets/transaction_header.dart [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+    this.onSeeAllPressed,
+  });
+
+  @override
+  Widget build(BuildContext context) {
+    final theme = Theme.of(context);
+    return Padding(
+      padding: EdgeInsets.symmetric(
+        horizontal: theme.sizing.width.s4,
+        vertical: theme.sizing.width.s2,
```
Comment (quiteachaos) [inline lib/widgets/transaction_header.dart]: vertical: theme.sizing.height.s2

File: lib/widgets/transaction_header.dart [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+            (onSeeAllPressed != null) ? 'Transactions' : 'All Transactions',
+            style: theme.textStyle.titleRegular.copyWith(
+              color: theme.colors.contrastDark,
+            ),
+          ),
+          const SizedBox(width: 10),
+          Container(
+            padding: EdgeInsets.symmetric(
+                horizontal: theme.sizing.width.s2,
+                vertical: theme.spacing.width.s4),
```
Comment (quiteachaos) [inline lib/widgets/transaction_header.dart]: vertical: theme.sizing.height.s4,

File: lib/widgets/transaction_header.dart [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+      ),
+      child: Row(
+        children: [
+          Text(
+            (onSeeAllPressed != null) ? 'Transactions' : 'All Transactions',
+            style: theme.textStyle.titleRegular.copyWith(
+              color: theme.colors.contrastDark,
+            ),
+          ),
+          const SizedBox(width: 10),
```
Comment (quiteachaos) [inline lib/widgets/transaction_header.dart]: use theme; if 10 is not available try using 8 or 12 whichever suits better

File: lib/widgets/transaction_header.dart [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+                color: theme.colors.contrastMedium,
+              ),
+            ),
+          ),
+          const Spacer(),
+          if (onSeeAllPressed != null)
+            GestureDetector(
+              onTap: onSeeAllPressed,
+              child: Text(
+                'See All',
```
Comment (quiteachaos) [inline lib/widgets/transaction_header.dart]: use strings_constant
seeAllText

File: lib/widgets/transaction_header.dart (hunk 1 above)
Comment (NileshSingh9245) [reply lib/widgets/transaction_header.dart]: done

File: lib/widgets/transaction_header.dart (hunk 2 above)
Comment (NileshSingh9245) [reply lib/widgets/transaction_header.dart]: done

File: lib/widgets/transaction_header.dart (hunk 3 above)
Comment (NileshSingh9245) [reply lib/widgets/transaction_header.dart]: Done, used spacing and s10

File: lib/widgets/transaction_header.dart (hunk 4 above)
Comment (NileshSingh9245) [reply lib/widgets/transaction_header.dart]: done, by mistake it was pushed that way

**Review by quiteachaos — APPROVED** [review]

**Review by satyabratanayak — APPROVED** [review]

**Pull request merged**